### PR TITLE
Provide ability to whitelist puppet exit code.

### PR DESCRIPTION
**Pull request by ytsarev**
* New puppet_whitelist_exit_code configuration option
* Move remove_repo to cleanup section so it does not hide puppet exit
  code
* Relevant documentation

### CODE REVIEW

**Comment by ytsarev** [comment]
@grubernaut @neillturner So guys, I've initially overlooked existing `puppet_detailed_exitcodes` option. Now adding new one to be able to work with it.

**Comment by neillturner** [comment]
this looks ok

File: lib/kitchen/provisioner/puppet_apply.rb [hunk 1]
```diff
@@ -537,7 +541,7 @@ def run_command
             puppet_verbose_flag,
             puppet_debug_flag,
             puppet_logdest_flag,
-            remove_repo
```
Comment (grubernaut) [inline lib/kitchen/provisioner/puppet_apply.rb]: Was this intentional to leave?

Comment (ytsarev) [inline lib/kitchen/provisioner/puppet_apply.rb]: It is moved to proper place https://github.com/neillturner/kitchen-puppet/pull/112/files#diff-851e4fa01ea842ebe9366ac535ce73b0R423

Comment (grubernaut) [inline lib/kitchen/provisioner/puppet_apply.rb]: Awesome, LGTM

**Comment by neillturner** [comment]
+1

**Comment by denys-mazhar-rp** [comment]
Really waiting for this feature. Have you guys any plans for the future release where it'll be included?

**Comment by neillturner** [comment]
i will release this weekend no problem.

**Comment by denys-mazhar-rp** [comment]
Thanks!

**Pull request merged**